binderhub/base.py

```diff
@@ -55,6 +58,44 @@ def write_error(self, status_code, **kwargs):
             message=message,
         )
 
+    def _normalize_cookie(self, cookie):
+        """
+        Make sure the json cookie is normalized to have a know schema:
+        {
+            'known': [list of unique strings]
+            'default': [string found in known section]
+        }
+
+        We could (should ?) be stricter.
+
+        """
+
+        portal_address = self.settings['cannonical_address']
```

Comment:
A bit confusing to change name and use portal address. Perhaps:
`federation_site_address = self.settings['federation_site_address]`